src/main/java/org/broadinstitute/hellbender/tools/spark/sv/ContigAligner.java

```diff
@@ -0,0 +1,272 @@
+     * of the genome (with one primary and two supplementary alignment records), there will be two AssembledBreakpoint
+     * objects created, one to represent each junction between alignment regions:
+     *
+     * Example Contig:
+     * ACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTGCACTGACTG
+     * Alignment regions:
+     * |---------1:100-200------------|
+     *                                |-----------2:100-200------------------|
+     *                                                                       |----------3:100-200-----------------|
+     * Assmbled breakpoints:
+     * 1) links 1:100-200 to 2:100-200
+     * 2) links 2:100-200 to 3:100-200
+     *
+     * Inserted sequence contains portions of the contig that are aligned to neither region, and therefore may be inserted in
+     * the sample. For example, a translocation breakpoint with a microinsertion:
+     *
+     * Contig:
+     * ACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTG
+     * Alignment regions:
+     * |-----1:100-200-------|
+     *                         |-----2:100-200-----|
+     * Inserted sequence:
+     *  GA
+     *
+     * Homology represents ambiguity about the exact location of the breakpoint. For example, in this case one alignment
+     * region ends with "AC" and the next begins with AC, so we don't know if the AC truly belongs with the first or
+     * second alignment region.
+     *
+     * Contig:
+     * ACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTG
+     * Alignment regions:
+     * |-----1:100-200-------|
+     *                    |-----2:100-200----------|
+     * Homology:
+     *  AC
+     */
+    static class AssembledBreakpoint {
+        String contigId;
+        AlignmentRegion region1;
+        AlignmentRegion region2;
+        String insertedSequence;
+        String homology;
+
+        @Override
+        public String toString() {
+            final StringBuilder stringBuilder = new StringBuilder();
+            stringBuilder.append(contigId);
+            stringBuilder.append("\t");
+            stringBuilder.append(region1.toString());
+            stringBuilder.append("\t");
+            stringBuilder.append(region2.toString());
+            stringBuilder.append("\t");
+            stringBuilder.append(insertedSequence);
+            stringBuilder.append("\t");
+            stringBuilder.append(homology);
+            return stringBuilder.toString();
+        }
+    }
+
+    static class AlignmentRegion {
+
+        final Cigar cigar;
+        final boolean forwardStrand;
+        final SimpleInterval interval;
+        final int mqual;
+        final int startInContig;
+        final int endInContig;
+        final int contigLength;
+
+
+        public AlignmentRegion(final AlnRgn alnRgn) {
+            this.forwardStrand = alnRgn.getStrand() == '+';
+            this.cigar = TextCigarCodec.decode(alnRgn.getCigar());
```

Comment:
I'd consider reversing the cigar right here when it's initialized if we have a reverse strand mapping.  That simplifies getClippedBases, and will probably be more intuitive for users (who won't expect to have to consult the forwardStrand flag for correct cigar interpretation).  I.e., it'll be more BAM-like.